onadata/libs/utils/logger_tools.py

```diff
@@ -594,36 +594,21 @@ def create_instance(
 
         return DuplicateInstance()
 
-    try:
-        with transaction.atomic():
-            if isinstance(xml, bytes):
-                xml = xml.decode("utf-8")
-            instance = save_submission(
-                xform,
-                xml,
-                media_files,
-                new_uuid,
-                submitted_by,
-                status,
-                date_created_override,
-                checksum,
-                request,
-            )
-    except IntegrityError:
-        instance = get_first_record(
-            Instance.objects.filter(uuid=new_uuid, xform_id=xform.pk)
+    with transaction.atomic():
+        if isinstance(xml, bytes):
+            xml = xml.decode("utf-8")
+        instance = save_submission(
+            xform,
+            xml,
+            media_files,
+            new_uuid,
+            submitted_by,
+            status,
+            date_created_override,
+            checksum,
+            request,
         )
 
-        if instance:
-            attachment_names = [
-                a.media_file.name.split("/")[-1]
-                for a in Attachment.objects.filter(instance=instance)
-            ]
-            media_files = [f for f in media_files if f.name not in attachment_names]
-            save_attachments(xform, instance, media_files)
-            instance.save()
-
```

Comment:
Is saving attachments no longer necessary?

Reply:
After testing and carefully looking at the code, in case of a duplicate submission, the code execution will never reach that code since the attachment processing will be handled in the `if existing_instance and (new_uuid or existing_instance.xform.has_start_time)` block, a few lines above the block I removed.

I then updated [Attachment.objects.get_or_create inside save_attachments method](https://github.com/onaio/onadata/pull/2860/files#diff-76a3d3d03f02ac0438c8082454afcf25cea152b40d7430deb3130c7b00d6023dR444) and excluded the file object from the uniqueness check since it's guaranteed the file object references will always be different